File: fibonacci-number/fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        x = 1;
        y = 0;
        f = x + y
        out = 0 ;
        if(n==0 | n==1):
            out = n;
        elif n>=2:
            for i in range(2,n):
                y = x
                x = f
                f = x + y
            out = f
            
        return out

File: fibonacci-number/fibonacci-number.java
class Solution {
    public int fib(int n) {
        return fibo(n);
    }
    
    public int fibo(int n){
        int out =0;
        if(n==0 || n==1){
            out = n;
        }else {
            out = fibo(n-1) + fibo(n-2);
        }
        return out;
    }
}